cijoe/scripts/bench_reporter.py

```diff
@@ -14,8 +14,156 @@
 ------------------
 """
 import errno
+from datetime import date
 from pathlib import Path
 from shutil import copyfile
+from typing import Dict, Iterator, List, Tuple
+
+import jinja2
+
+
+def lazy_read(path: Path) -> Iterator[str]:
+    with path.open("r") as lines:
+        for line in lines:
+            yield line.strip()
+
+
+def get_host(path: Path) -> str:
+    with path.open("r") as output:
+        return output.readline().strip()
+
+
+def get_linux(path: Path) -> str:
+    linux_version = ""
+    with path.open("r") as lines:
+        for line in (line.strip() for line in lines):
+            if "Description" in line.split(":")[0]:
+                linux_version = line.split(":")[1]
+                break
+
+    return linux_version
+
+
+def get_kernel(path: Path) -> str:
+    with path.open("r") as output:
+        return output.readline().strip()
+
+
+def read_sysinfo(sysinfo_path: Path) -> Tuple[str, str]:
+    hostname = ""
+    linux_version = ""
+    kernel_version = ""
+
+    for state_path in sysinfo_path.glob("*.state"):
+        with state_path.open("r") as state:
+            for line in (line.strip() for line in state):
+                output_path = sysinfo_path / state_path.name.replace(
+                    ".state", ".output"
+                )
+                right, left = line.strip().split(":")
+                if "cmd" in right:
+                    if "hostname" in left:
```

Comment:
It could be an idea to define your strings as constant, as an example HOSTNAME = "hostname"

Reply:
I will do this for the ones that are repeated